src/apply.c

```diff
@@ -199,13 +199,24 @@ static int apply_hunk(
 
 	for (i = 0; i < hunk->line_count; i++) {
 		size_t linenum = hunk->line_start + i;
+		git_diff_line *next;
 		git_diff_line *line = git_array_get(patch->lines, linenum);
 
 		if (!line) {
 			error = apply_err("preimage does not contain line %"PRIuZ, linenum);
 			goto done;
 		}
 
+		/* lookahead for eofnl */
+		if (i < hunk->line_count - 1) {
+			next = git_array_get(patch->lines, linenum + 1);
+			if (next->origin == GIT_DIFF_LINE_CONTEXT_EOFNL ||
+					next->origin == GIT_DIFF_LINE_DEL_EOFNL ||
+					next->origin == GIT_DIFF_LINE_ADD_EOFNL) {
```

Comment:
Nit: indentation is off here. If conditionals span multiple lines, then the conditions should be indented by four spaces relative to the `if`, only:

```
if (next->origin == GIT_DIFF_LINE_CONTEXT_EOFNL ||
    next->origin == GIT_DIFF_LINE_DEL_EOFNL ||
    next->origin == GIT_DIFF_LINE_ADD_EOFNL) {
        ...
}
```

Reply:
But the project is set to indent with TAB

Reply:
Yeah. We had some misunderstandings around that in the past, but our rule is to follow git.git and thus linux.git's style. They use 8-space tabs in general, but indent multiline conditionals with four spaces. I've never used editorconfig, but if it is possible to specify this rule then we should certainly fix it.